.github/workflows/spelling.yml

```
@@ -0,0 +1,30 @@
+name: Spell checking
+on:
+  push:
+    branches:
+      - "**"
+    tags-ignore:
+      - "**"
+  pull_request_target:
+  issue_comment:
+    types: [created]
```

If I understand correctly this will trigger on pushes to all branches (because of `"**"`), just remove `branches`?
Why do you ignore tags? All other workflows work just fine, I'd suggest to just remove it (unless I'm missing something).
Why `pull_request_target` and not just `pull_request`?
```yml
  issue_comment:
    types: [created]
```
Surely we don't need to trigger the workflow on every comment on every issue.

I think this would be enough:
```suggestion
on: [push, pull_request]
```